homeassistant/components/arduino.py

```diff
@@ -0,0 +1,183 @@
+        if mode == 'analog' and direction == 'in':
+            self._board.set_pin_mode(pin,
+                                     self._board.INPUT,
+                                     self._board.ANALOG)
+            self._board.enable_analog_reporting(pin)
+        elif mode == 'analog' and direction == 'out':
+            self._board.set_pin_mode(pin,
+                                     self._board.OUTPUT,
+                                     self._board.ANALOG)
+        elif mode == 'digital' and direction == 'in':
+            self._board.set_pin_mode(pin,
+                                     self._board.OUTPUT,
+                                     self._board.DIGITAL)
+            self._board.enable_digital_reporting(pin)
+        elif mode == 'digital' and direction == 'out':
+            self._board.set_pin_mode(pin,
+                                     self._board.OUTPUT,
+                                     self._board.DIGITAL)
+        elif mode == 'pwm':
+            self._board.set_pin_mode(pin,
+                                     self._board.OUTPUT,
+                                     self._board.PWM)
+
+    def digital_out_high(self, pin):
+        """ Sets a given digital pin to high. """
+        self._board.digital_write(pin, 1)
+
+    def digital_out_low(self, pin):
+        """ Sets a given digital pin to low. """
+        self._board.digital_write(pin, 0)
+
+    def get_firmata(self):
+        """ Return the version of the Firmata firmware. """
+        return self._board.get_firmata_version()
+
+    def disconnect(self):
+        """ Disconnects the board and closes the serial connection. """
+        self._board.close
+
+
+class ArduinoDeviceABC(ToggleEntity):
+    """ Abstract Class for an Arduino board within Home Assistant. """
+
+    _states = []
+    _domain = None
+    _name = None
+
+    def __init__(self, name, pin):
+        # Setup properties
+        self._pin = pin
+        self._name = name  or DEVICE_DEFAULT_NAME
+        self._state = STATE_OFF
+        self._value = None
+
+    @property
+    def domain(self):
+        """ Returns the domain of the entity. """
+        return self._domain
+
+    @property
+    def is_on(self):
+        """ Returns True if switch is on. """
+        return self._state == STATE_ON
+
+    @property
+    def should_poll(self):
+        """ Tells Home Assistant not to poll this entity. """
+        return False
+
+    @property
+    def name(self):
+        """ Returns the name of the pin. """
+        return self._name
+
+    @property
+    def state(self):
+        """ Returns the state of the device if any. """
+        return self._state
+
+    def on_update(self, event):
+        """ Handles the update received event. """
+        self.update_ha_state()
+
+    def turn_on(self):
+        """ Turns the pin to high/on. """
+        if self.domain is not 'sensor':
+            self._state = STATE_ON
+            BOARD.set_mode(self._pin, 'out', 'digital')
```

Comment:
I am not at all familiar with Arduino so maybe I'm wrong. Wouldn't calling this method in the constructor be enough?